server/util/db/db.go

```diff
@@ -347,8 +361,16 @@ func GetConfiguredDatabase(c *config.Configurator, hc interfaces.HealthChecker)
 	go statsRecorder.poll(statsPollInterval)
 
 	if err := maybeRunMigrations(dialect, primaryDB); err != nil {
+		if *autoMigrateDBAndExit {
+			log.Fatalf("Database auto-migration failed: %s", err)
+		}
+
 		return nil, err
 	}
+	if *autoMigrateDBAndExit {
+		log.Infof("Database migration completed. Exiting due to --auto_migrate_db_and_exit.")
+		os.Exit(0)
+	}
```

Comment:
Could we clean some of this logic up by doing all the flag-based switching at the outer level -- instead of at the outer level and inside of maybeRunMigrations? Something like:


if *autoMigrateDBAndExit {
 if err := runMigrations(dialect, primaryDB); err != nil {
   log.Fatalf("Database auto-migration failed: %s", err)
 }
 log.Infof("Database migration completed. Exiting due to --auto_migrate_db_and_exit.")
 os.Exit(0)
}

if *autoMigrateDB {
  if err := runMigrations(dialect, primaryDB); err != nil {
    return err
  }
}

Reply:
Much cleaner, done